hubblestack/extmods/grains/cloud_details.py

```diff
@@ -40,17 +43,27 @@ def _get_aws_details():
         token_request = requests.put(token_url, headers=ttl_header, timeout=3, proxies=proxies)
         token = token_request.text
         aws_token_header = {'X-aws-ec2-metadata-token': token}
-        res = requests.get('http://169.254.169.254/latest/dynamic/instance-identity/document',
-                            headers=aws_token_header, timeout=3, proxies=proxies).json()
-        aws['cloud_account_id'] = res.get('accountId', 'unknown')
-
-        # AWS account id is always an integer number
-        # So if it's an aws machine it must be a valid integer number
-        # Else it will throw an Exception
-        int(aws['cloud_account_id'])
-        aws['cloud_instance_id'] = requests.get('http://169.254.169.254/latest/meta-data/instance-id',
-                                                headers=aws_token_header, timeout=3,
-                                                proxies=proxies).text
+        response = requests.get('http://169.254.169.254/latest/dynamic/instance-identity/document',
+                            headers=aws_token_header, timeout=3, proxies=proxies)
+        if response.status_code == requests.codes.ok:
+            res = response.json()
+            aws['cloud_account_id'] = res.get('accountId', 'unknown')
+
+            # AWS account id is always an integer number
+            # So if it's an aws machine it must be a valid integer number
+            # Else it will throw an Exception
+            int(aws['cloud_account_id'])
+        else:
+            log.error("Error while fetching AWS account id. Got status code: %s " % (response.status_code))
+            raise ValueError("Error while fetching AWS account id. Got status code: %s " % (response.status_code))
```

Comment:
1. You can move this log.error in catch() block. It will remove this duplicate error string at two places.
    Same for code at line 65
2. In error message, put account id as well.